### PR TITLE
Add a face alignment based tracker

### PR DESCRIPTION
**Pull request by DaWelter**
Follow up of issue https://github.com/opentrack/opentrack/issues/1247.

This PR adds a face alignment / neural network based tracker which works only on RGB video without markers.

Models were generated with code from
https://github.com/DaWelter/neuralnet-tracker-traincode/releases/tag/v0.1

I also propose the neuralnet-tracker-traincode repo to be forked by the opentrack orga

### CODE REVIEW

File: tracker-neuralnet/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    otr_module(tracker-neuralnet)
+    target_include_directories(${self} SYSTEM PUBLIC 
+        ${OpenCV_INCLUDE_DIRS} "${ONNX_INCLUDE_DIR}")
+    target_link_libraries(${self} 
+        opentrack-cv "${SDK_ONNX_LIBPATH}" opencv_imgproc opencv_core 
+        opencv_imgcodecs opencv_calib3d)
+
+    install(
+        FILES "models/head-localizer.onnx" "models/head-pose.onnx"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/${opentrack-i18n}/../models/"
```
Comment (sthalik) [inline tracker-neuralnet/CMakeLists.txt:19]: Use `DESTINATION "${opentrack-libexec}/models"`.

File: tracker-neuralnet/ftnoir_tracker_neuralnet.h [hunk 2]
```diff
@@ -0,0 +1,231 @@
+/* Copyright (c) 2013-2015 Stanislaw Halik <sthalik@misaki.pl>
```
Comment (sthalik) [inline tracker-neuralnet/ftnoir_tracker_neuralnet.h:1]: Add your own copyright statement, remove mine.

File: tracker-neuralnet/ftnoir_tracker_neuralnet.cpp [hunk 3]
```diff
@@ -0,0 +1,917 @@
...
+}
+
+
+int PoseEstimator::find_input_intensity_90_pct_quantile() const
+{
+    const int channels[] = { 0 };
+    const int hist_size[] = { 255 };
+    float range[] = { 0, 256 };
+    const float* ranges[] = { range };
+    cv::Mat hist; // TODO: Preallocate, but whatever I tried, I ran into asserts in opencv ...
```
Comment (sthalik) [inline tracker-neuralnet/ftnoir_tracker_neuralnet.cpp:267]: I've had the same problem, the solution was adding a member variable and letting opencv allocate it the first time. The array is always the same size and type regardless of user-specified options.

File: tracker-neuralnet/ftnoir_tracker_neuralnet.cpp [hunk 4]
```diff
@@ -0,0 +1,917 @@
...
+    {
+        const int hz = enum_to_fps(k);
+        const QString name = (hz == 0) ? tr("Default") : QString::number(hz);
+        ui.cameraFPS->addItem(name, k);
+    }
+}
+
+
+neuralnet_dialog::neuralnet_dialog() :
+    trans_calib(1, 2)
```
Comment (sthalik) [inline tracker-neuralnet/ftnoir_tracker_neuralnet.cpp:764]: Initialize inline in the header file:

```c++
Foo bar{1, 2, 3};
```

File: tracker-neuralnet/ftnoir_tracker_neuralnet.cpp [hunk 5]
```diff
@@ -0,0 +1,917 @@
...
+    assert (input_mat.ptr(0) == p);
+    assert (!input_mat.empty() && input_mat.isContinuous());
+    assert (input_mat.cols == input_img_width && input_mat.rows == input_img_height);
+
+    const char* input_names[] = {"x"};
+    const char* output_names[] = {"logit_box"};
+
+    //Timer t_; t_.start();
+
+    const auto nt = omp_get_num_threads();
```
Comment (sthalik) [inline tracker-neuralnet/ftnoir_tracker_neuralnet.cpp:212]: openmp should be looked up using the build system.

**Review by sthalik — CHANGES_REQUESTED** [review]
It's fine except for the copyright statement.

File: tracker-neuralnet/ftnoir_tracker_neuralnet.cpp (hunk 3 above)
Comment (DaWelter) [reply tracker-neuralnet/ftnoir_tracker_neuralnet.cpp]: Hm, doesn't work for me. But doesn't matter. I guess trying anything overly clever here amounts to premature optimization anyway.

**Comment by sthalik** [comment]
LGTM. 👍

**Review by sthalik — APPROVED** [review]

**Comment by DaWelter** [comment]
Let me try it on windows as well. Afterwards I will squash the commits.

**Comment by DaWelter** [comment]
Thanks for checking it out @sthalik . I just squashed two commits. From my side it is ready.

**Comment by GO63-samara** [comment]
Why merged into `opentrack:unstable` ?
Maybe better in `opentrack:master` ?

**Comment by sthalik** [comment]
@GO63-samara good catch, thanks. PR was against unstable.

**Comment by DaWelter** [comment]
Oh apologies. I figured `unstable` was the right place. What is the difference anyway?

**Pull request merged**